### PR TITLE
setup ossf scorecard and codql workflows

### PR DESCRIPTION
**Pull request by mmorel-35**
Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

**Comment by mmorel-35** [comment]
cc @phlax

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: 2
+updates:
+  - package-ecosystem: github-actions
```
Comment (phlax) [inline .github/dependabot.yml:3]: indents!

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language:
+        - 'go'
+        - 'javascript-typescript'
+        - 'python'
```
Comment (phlax) [inline .github/workflows/codeql.yml:32]: unreltedly - ive been thinking we should add some other langs in the envoy setup

File: .github/workflows/csds-client.yaml [hunk 3]
```diff
@@ -1,4 +1,8 @@
 name: Build and test
+
+permissions:
+  contents: read
+
 on:
   - push
```
Comment (phlax) [inline .github/workflows/csds-client.yaml:7]: i think we can fix these indents - but happy to do it separately

**Review by phlax — COMMENTED** [review]
usual gripes 8)

**Review by phlax — APPROVED** [review]
lgtm, thanks @mmorel-35

**Pull request merged**